packages/react/src/ElementsInput/ElementsInput.tsx

```diff
@@ -0,0 +1,138 @@
+import { Stack } from '@mantine/core';
+import { InternalSchemaElement, TypedValue, getPathDisplayName } from '@medplum/core';
+import { useMemo, useState } from 'react';
+import { CheckboxFormSection } from '../CheckboxFormSection/CheckboxFormSection';
+import { DEFAULT_IGNORED_NON_NESTED_PROPERTIES, DEFAULT_IGNORED_PROPERTIES } from '../constants';
+import { FormSection } from '../FormSection/FormSection';
+import { setPropertyValue } from '../ResourceForm/ResourceForm.utils';
+import { getValueAndTypeFromElement } from '../ResourcePropertyDisplay/ResourcePropertyDisplay.utils';
+import { ResourcePropertyInput } from '../ResourcePropertyInput/ResourcePropertyInput';
+import { OperationOutcome } from '@medplum/fhirtypes';
+
+const EXTENSION_KEYS = ['extension', 'modifierExtension'];
+
+export interface ElementsInputProps {
+  type: string | undefined;
+  elements: { [key: string]: InternalSchemaElement };
+  defaultValue: any;
+  outcome: OperationOutcome | undefined;
+  onChange: ((value: any) => void) | undefined;
+}
+
+export function ElementsInput(props: ElementsInputProps): JSX.Element {
+  const { elements } = props;
+  const [value, setValue] = useState<any>(props.defaultValue ?? {});
+
+  const fixedProperties = useMemo(() => {
+    const result: { [key: string]: InternalSchemaElement & { fixed: TypedValue } } = Object.create(null);
+    for (const [key, property] of Object.entries(elements)) {
+      if (property.fixed) {
+        result[key] = property as any;
+      }
+    }
+    return result;
+  }, [elements]);
+
+  function setValueWrapper(newValue: any): void {
+    for (const [key, prop] of Object.entries(fixedProperties)) {
+      setPropertyValue(newValue, key, key, prop, prop.fixed.value);
+    }
+    console.log('ElementsInput', newValue);
+    setValue(newValue);
+    if (props.onChange) {
+      props.onChange(newValue);
+    }
+  }
+
+  return (
+    <Stack style={{ flexGrow: 1 }}>
+      {Object.entries(elements).map(([key, element]) => {
+        if (!element.type) {
+          return null;
+        }
+
+        if (element.max === 0) {
+          return null;
+        }
+
+        // mostly for Extension.url
+        if (key === 'url' && element.fixed) {
+          return null;
+        }
+
+        if (EXTENSION_KEYS.includes(key)) {
+          // TODO{mattlong} verify the following comment is accurate
+          // an extension property without slices has no sub-extensions nor content
```

Comment:
🤔 

Globally?  No, that is not true. Many resources use nested extensions where the outer extension has a fully qualified URL, and the inner extensions have a friendly name.

In the context of FHIR Profiles?  That might be true?  @mattwiller  do you know?

Reply:
Right, but isn't the only way to specify the nested extension by using a slice on the outer extension?

Reply:
e.g. `Extension.extension:ethnicity`?

Reply:
> the only way to specify the nested extension by using a slice on the outer extension

I don't know, but that seems likely.

I think I missed the key term here, "extension property", meaning a property defined in a FHIR Profile?

In that case, yes I believe the sentence "an extension property without slices has no sub-extensions nor content" is true.